Add a configuration file and a quiet mode

diml opened the pull request:
Once we don't stop at the first error, the output becomes messy. This PR adds display modes:

- `quiet`: only print errors
- `progress`: interactive display showing the number of rules processed so far and the number of parallel jobs running. This is the new default
- `short`: current default
- `verbose`: same as current `--verbose`

The display mode can be selected via the `--display` command line argument, or via the configuration file `~/.config/dune/config`.

TODO:
- [ ] Document display modes in the manual
- [ ] Document the configuration file in the manual

rgrinberg commented on src/scheduler.ml:
```diff
@@ -79,34 +107,58 @@ let wait_for_process pid =
-     fiber)
+    (Fiber.fork_and_join_unit
+       (fun () -> go_rec t)
+       (fun () -> fiber))
```
Previously, this change was simply omitted, right? I assume this is just for perf.

diml replied:
Yes, it was a mistake. This showed up after this patch as sometimes the status line wasn't cleaned up at the end

rgrinberg commented on bin/main.ml:
```diff
@@ -85,6 +84,20 @@ module Main = struct
+  include Jbuilder.Scheduler
+
+  let go ?log ~common fiber =
+    Scheduler.go ?log ~config:common.config fiber
```
I wonder if we should just initialize the log here from `~common`. I think that's what we always do anywhere.

diml replied:
indeed, although I think there is a case where we call the scheduler twice

rgrinberg commented:
Also it would be nice to enumerate the display options in `--help` as well. It's a bit of overkill to view the manual just for this.

diml commented:
> Seems to work well for me and the new output is definitely an improvement. Other candidates for this config file are things like the diff cmd.

Indeed. We could also allow to customize the status line:

```scheme
(status_line "foo: ${todo}, jobs: ${jobs}")
```

One things I'm going to add straight after that is:

```scheme
(jobs 20)
```

> Also it would be nice to enumerate the display options in --help as well. It's a bit of overkill to view the manual just for this.

indeed

diml commented:
I added the doc. I was finding that the man page was getting big, so I added `jbuilder help config`. I haven't updated the manual yet, though I find it painful to document things twice...
